controllers/userController.js

```diff
@@ -147,4 +148,41 @@ const deleteUserAddresses = async (req, res) => {
     res.status(500).send({ message: err })
   }
 }
-module.exports = { userRegister, login, getUserData, deleteUserData, limitUsersData, userAddress, deleteUserAddresses }
+
+const passwordResetToken = async (req, res) => {
+  const { username } = req.body
+  try {
+    const userData = await userModel.findOne({ where: { username } })
+    if (userData) {
+      const passwordResetToken = jwt.sign({
+        data: `${userData.username} ${userData.password}`
+      }, process.env.jwtSecKey);
+      await userModel.update({ passwordResetToken: passwordResetToken }, { where: { username } })
+      res.status(200).send(passwordResetToken)
+
+    } else {
+      res.status(404).send({ message: "user not found, please check the user id" })
+    }
+
+  } catch (err) { res.status(500).send(err) }
+
+}
+const verifyResetPasswordToken = async (req, res) => {
+  const resetToken = req.params.passwordResetToken
+  const { username, new_password } = req.body
+  try {
+    const userData = await userModel.findOne({ where: { username } })
+    const verifyToken = jwt.verify(resetToken, process.env.jwtSecKey, { expiresIn: 60 * 60 })
+    if (userData.passwordResetToken === resetToken && verifyToken) {
```

Comment:
Here first you verify the token after that check password is coming or not, if password not present then take give a error, or if available then create a new password and update in the password field